feat: Introduced simple narbar using tailwind

mehul1011 opened the pull request:
Introduced a simple navbar

Let me know if it suits you, I am starting off my open source journey, making some contributions to angular-based projects, and found your to be very interesting, also if you want to take this project forward we can get in touch or discuss it here by raising some issues for the changes required.

Thanks :)

mehul1011 committed:
feat: Introduced simple narbar using tailwind

mehul1011 committed:
feat + fix: introduced focus on textBox upon component being loaded, and fixed source code btn on larger screens

yagizhanNY commented:
Hi @mehul1011, thanks for your contribution request. 

Every contribution is welcome of course :)

I have checked your changes, it's looks perfect, but I have a comment.

- App is using dark theme so maybe you can use different background color for navbar which more fits for dark theme. For example;

![image](https://user-images.githubusercontent.com/41087670/229855028-710ec6cd-5fec-430a-873c-8feecd868951.png)

yagizhanNY commented on src/app/chat/chat.component.html:
```diff
@@ -1,4 +1,4 @@
-<div class="mb-32">
+<div class="mt-24">
```
you have to keep `mb-32`. Otherwise, conversation will get under to message input component when the conversation is too long.

mehul1011 replied:
The thing is when i made the navbar sticky the Chat content was going under the navbar, can you check once when you are available, this may be due to position absolute internally being applied in which the content is disturbed and the Chat content is not aware that navbar is actually present there.

yagizhanNY replied:
I mean you have to use both :)

`class="mb-32 mt-24"`

mehul1011 commented:
Hmm, that would be much better, will push it once done.

mehul1011 commented:
> Hi @mehul1011, thanks for your contribution request.
> 
> Every contribution is welcome of course :)
> 
> I have checked your changes, it's looks perfect, but I have a comment.
> 
> * App is using dark theme so maybe you can use different background color for navbar which more fits for dark theme. For example;
> 
> ![image](https://user-images.githubusercontent.com/41087670/229855028-710ec6cd-5fec-430a-873c-8feecd868951.png)

You may like this, let me know 
![Screenshot 2023-04-04 at 9 58 31 PM](https://user-images.githubusercontent.com/42713198/229857652-05831a77-03b1-4cab-ae04-6b50c53179cb.png)

mehul1011 commented:
> > Hi @mehul1011, thanks for your contribution request.
> > Every contribution is welcome of course :)
> > I have checked your changes, it's looks perfect, but I have a comment.
> > 
> > * App is using dark theme so maybe you can use different background color for navbar which more fits for dark theme. For example;
> > 
> > ![image](https://user-images.githubusercontent.com/41087670/229855028-710ec6cd-5fec-430a-873c-8feecd868951.png)
> 
> You may like this, let me know ![Screenshot 2023-04-04 at 9 58 31 PM](https://user-images.githubusercontent.com/42713198/229857652-05831a77-03b1-4cab-ae04-6b50c53179cb.png)

![Screenshot 2023-04-04 at 10 05 49 PM](https://user-images.githubusercontent.com/42713198/229858682-a9ee59fa-c382-49a0-bd18-bb85d64b59f5.png)

or we can have that border-shadow design too, that you were suggesting.

yagizhanNY commented:
What do you think about this color `#232323`? This might be much better.

I am usually using Figma for decide to colors. I found this color from this template https://www.figma.com/file/Vf0AiVgExPH7tiPpzWfzpk/ChatGPT-(Community)?node-id=5-104&t=LMEKt1oXjCGI2pIf-0

![image](https://user-images.githubusercontent.com/41087670/229988790-d1648805-5a2d-430c-9c12-0e73c812f00a.png)

Also it would be nice if the style of the button is same as the template.

Thanks for your effort :)

mehul1011 commented:
Sure @yagizhanNY 
I have checking on it, having some local setup issue for running npm etc

mehul1011 commented:
> What do you think about this color `#232323`? This might be much better.
> 
> I am usually using Figma for decide to colors. I found this color from this template https://www.figma.com/file/Vf0AiVgExPH7tiPpzWfzpk/ChatGPT-(Community)?node-id=5-104&t=LMEKt1oXjCGI2pIf-0
> 
> ![image](https://user-images.githubusercontent.com/41087670/229988790-d1648805-5a2d-430c-9c12-0e73c812f00a.png)
> 
> Also it would be nice if the style of the button is same as the template.
> 
> Thanks for your effort :)

I think it would be better to keep dark mode first, especially the chat, if i were a user i would not visit a site with light mode 🤣 
Just personal preference though, let me know, we can have a bit lighter mode than that chat, and have a box-shadow or something.

mehul1011 commented:
> > What do you think about this color `#232323`? This might be much better.
> > I am usually using Figma for decide to colors. I found this color from this template https://www.figma.com/file/Vf0AiVgExPH7tiPpzWfzpk/ChatGPT-(Community)?node-id=5-104&t=LMEKt1oXjCGI2pIf-0
> > ![image](https://user-images.githubusercontent.com/41087670/229988790-d1648805-5a2d-430c-9c12-0e73c812f00a.png)
> > Also it would be nice if the style of the button is same as the template.
> > Thanks for your effort :)
> 
> I think it would be better to keep dark mode first, especially the chat, if i were a user i would not visit a site with light mode 🤣 Just personal preference though, let me know, we can have a bit lighter mode than that chat, and have a box-shadow or something.

@yagizhanNY Here's an example with a shadow, but if we have to go with 2 column layout we can do that.
![Screenshot 2023-04-05 at 9 33 40 PM](https://user-images.githubusercontent.com/42713198/230139504-bacb99bd-9d1a-4e7c-a673-7340354a136b.png)

mehul1011 committed:
fix: Navbar to have consitent margins so that content doesn't overflow

mehul1011 commented:
I have updated the branch with mb & mt that you suggested.

yagizhanNY commented:
I mean we can use this color `#232323` for navbar background in dark mode. I am agree with you about light mode :)

mehul1011 committed:
fix: using trueGray color == #232323 for navbar

mehul1011 commented:
> `#232323`

For your reference
<img width="1440" alt="Screenshot 2023-04-05 at 11 41 16 PM" src="https://user-images.githubusercontent.com/42713198/230168061-d582386a-99ab-4383-9f71-e06fb9a64115.png">

yagizhanNY commented:
@mehul1011 I changed something. 

- You don't need to use fixed on navbar. It will always at top of the screen because of location the navbar component. I removed that.
- I removed the `mt-24` class from navbar. It is not fixed anymore so we don't need this class.

mehul1011 commented:
> @mehul1011 I changed something. 
> 
> - You don't need to use fixed on navbar. It will always at top of the screen because of location the navbar component. I removed that.
> - I removed the `mt-24` class from navbar. It is not fixed anymore so we don't need this class.

Ah okay, thanks for updating.
